src/render/draw_circle.js

```diff
@@ -45,8 +44,6 @@ function drawCircles(painter: Painter, sourceCache: SourceCache, layer: CircleSt
             gl.uniform2fv(program.uniforms.u_extrude_scale, painter.transform.pixelsToGLUnits);
         }
 
-        gl.uniform1f(program.uniforms.u_devicepixelratio, browser.devicePixelRatio);
```

Comment:
Why is this no longer necessary?

Reply:
It's an unused leftover — we now define DEVICE_PIXEL_RATIO as a shader constant in the shader generation code.